test/lib/PlacesTestUtils.js

```diff
@@ -24,22 +26,69 @@ const PlacesTestUtils = Object.freeze({
    *            [optional] visitDate: visit date in microseconds from the epoch
    *            [optional] referrer: nsIURI of the referrer for this visit
    *          }
-   *
-   * @return {Promise}
-   *            resolves When all visits have been added successfully.
-   *            rejects JavaScript exception.
    */
   addVisits: Task.async(function*(placeInfo) {
-    let promise = new Promise((resolve, reject) => {
-      let places = [];
-      if (placeInfo instanceof Ci.nsIURI) {
-        places.push({uri: placeInfo});
-      } else if (Array.isArray(placeInfo)) {
-        places = places.concat(placeInfo);
-      } else {
-        places.push(placeInfo);
+    let places = [];
+
+    if (placeInfo instanceof Ci.nsIURI) {
+      places.push({uri: placeInfo});
+    } else if (Array.isArray(placeInfo)) {
+      places = places.concat(placeInfo);
+    } else {
+      places.push(placeInfo);
+    }
+
+    // Keep track of the number of rows for final validation
+    function* getPlacesRows() {
```

Comment:
Ooh, what is "yield*" do? I've never seen that before? 

Sent from my iPhone

> On 11 Feb 2016, at 8:32 AM, Olivier Yiptong <notifications@github.com> wrote:
> 
> In test/lib/PlacesTestUtils.js:
> 
> > -      } else if (Array.isArray(placeInfo)) {
> > -        places = places.concat(placeInfo);
> > -      } else {
> > -        places.push(placeInfo);
> > +    let places = [];
> > +
> > +    if (placeInfo instanceof Ci.nsIURI) {
> > +      places.push({uri: placeInfo});
> > +    } else if (Array.isArray(placeInfo)) {
> > +      places = places.concat(placeInfo);
> > +    } else {
> > +      places.push(placeInfo);
> > +    }
> > +
> > +    // Keep track of the number of rows for final validation
> > +    function* getPlacesRows() {
> The original code was definitely wrong. I got a chance to look at it with @mzhilyaev and it looks like i should've used yield* instead of yield
> 
> —
> Reply to this email directly or view it on GitHub.
>